sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -387,8 +284,8 @@ export class MessageSender extends LinkEntity {
               this.name,
               delivery.id
             );
+            return resolve();
           } catch (error) {
-            removeListeners();
             return reject(error);
```

Comment:
Do we need to translate the error and look at `error.innerError`?
I noticed this was done in event hubs:
https://github.com/Azure/azure-sdk-for-js/pull/4637/files#diff-74cfdc5dbf644d05f5c8c6d4d5ca7f3eR647

There was some more context here: https://github.com/Azure/azure-sdk-for-js/issues/3375#issuecomment-517844630